### ISSUE TITLE
Can't run PowerShell on Ubuntu 20.04 arm64 due to a problem with the 'Microsoft.ApplicationInsights' assembly

### ISSUE THREAD

**Issue by damonbarry** [PowerShell/PowerShell#19055]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

On a fresh Ubuntu 20.04 arm64 VM, I installed the .NET 6.0 SDK (version 6.0.405). Then I took the following steps:

```bash
curl -sSL 'https://pwshtool.blob.core.windows.net/tool/7.2.9/PowerShell.Linux.arm64.7.2.9.nupkg' -o 'PowerShell.Linux.arm64.7.2.9.nupkg'
sudo mkdir -p /usr/share/powershell
sudo dotnet tool install --add-source . --tool-path /usr/share/powershell --version '7.2.9' PowerShell.Linux.arm64
sudo dotnet nuget locals all --clear
sudo ln -s /usr/share/powershell/pwsh /usr/bin/pwsh
sudo chmod 755 /usr/share/powershell/pwsh
pwsh
```

### Expected behavior

```console
`pwsh` should open a PowerShell session. And, in fact, it works as expected with previous versions, e.g., https://pwshtool.blob.core.windows.net/tool/7.2.8/PowerShell.Linux.arm64.7.2.8.nupkg.
```

### Actual behavior

```console
azureuser@my-arm-vm:~$ pwsh
PowerShell 7.2.9
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

Process terminated. Could not load file or assembly 'Microsoft.ApplicationInsights, Version=2.21.0.429, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. An attempt was made to load a program with an incorrect format.

   at System.Environment.FailFast(System.String, System.Exception)
   at Microsoft.PowerShell.UnmanagedPSEntry.Start(System.String[], Int32)
   at Microsoft.PowerShell.ManagedPSEntry.Main(System.String[])
System.BadImageFormatException: Could not load file or assembly 'Microsoft.ApplicationInsights, Version=2.21.0.429, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. An attempt was made to load a program with an incorrect format.

File name: 'Microsoft.ApplicationInsights, Version=2.21.0.429, Culture=neutral, PublicKeyToken=31bf3856ad364e35'
   at Microsoft.PowerShell.Telemetry.ApplicationInsightsTelemetry.SendPSCoreStartupTelemetry(String mode)
   at Microsoft.PowerShell.ConsoleHost.Start(String bannerText, String helpText)
   at Microsoft.PowerShell.UnmanagedPSEntry.Start(String[] args, Int32 argc)Aborted (core dumped)
```

### Error details

```console
N/A
```

### Environment data

```powershell
N/A
```

**Comment by rhubarb-geek-nz** [comment]
I use PowerShell from the tar. For example

`$ curl --silent --location https://github.com/PowerShell/PowerShell/releases/download/v7.2.9/powershell-7.2.9-linux-arm64.tar.gz | tar tvfz - | grep Microsoft.ApplicationInsights.dll
`

My installation of 7.3.1 on ARM64 from the tar in a VM works well

```
$ dpkg --print-architecture
arm64
$ grep VERSION_ID /etc/os-release 
VERSION_ID="20.04"
$ ls -ld /usr/bin/pwsh 
lrwxrwxrwx 1 root root 32 Nov 23 04:29 /usr/bin/pwsh -> /opt/microsoft/powershell/7/pwsh
$ ls -ld /opt/microsoft/powershell/7/*Microsoft.ApplicationInsights*
-rw-r--r-- 1 root root 1082792 Nov 23 04:11 /opt/microsoft/powershell/7/Microsoft.ApplicationInsights.dll
-rw-r--r-- 1 root root  379246 Nov 23 03:35 /opt/microsoft/powershell/7/Microsoft.ApplicationInsights.ni.r2rmap
$ pwsh --version
PowerShell 7.3.1
```

**Comment by rhubarb-geek-nz** [comment]
```
$ wget https://github.com/PowerShell/PowerShell/releases/download/v7.2.9/powershell-7.2.9-linux-arm64.tar.gz
tar xfz powershell-7.2.9-linux-arm64.tar.gz 
$ ./pwsh
PowerShell 7.2.9
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

PS /home/rogerb/work> 
```

**Comment by rhubarb-geek-nz** [comment]
That said, the permissions in the 7.3.* linux tar are wrong and should be fixed. [File permissions in linux tar are incorrect](https://github.com/PowerShell/PowerShell/issues/18899)

**Comment by isra-fel** [comment]
Met the same error on a MacOS 11 VM. PowerShell was 7.2.9, installed by dotnet tool

`dotnet tool install PowerShell --version 7.2.*`

**Comment by SteveL-MSFT** [comment]
@JamesWTruher we should not prevent startup if AppInsights can't be loaded

**Comment by BethanyZhou** [comment]
Hi @JamesWTruher , any workaround about this issue? Our daily test is blocked by this issue.

**Comment by JamesWTruher** [comment]
I'm sorry to say there's no work-around at present. The build of SMA has this as a referenced assembly so when the process starts up, that assembly will get loaded. It's the same for any referenced assembly. I'm still investigating.
 
There may be an issue with the dotnet installation - would it be possible to try the .zip installation?

**Comment by JamesWTruher** [comment]
@SteveL-MSFT 
> @JamesWTruher we should not prevent startup if AppInsights can't be loaded

yah - AFAIK, that's not really possible - it's a dependency of SMA, so if _any_ of the referenced assemblies can't be loaded, you would have the same issue.

**Comment by JamesWTruher** [comment]
@BethanyZhou it may be that the issue is with the `dotnet tool install` path. I replaced the AI assembly with the one from the distribution and the pwsh tool started just fine.
try `copy-item $PSHOME/Microsoft.ApplicationInsights.dll <dotnet install location>`
the dotnet tool install location on my mac was:
`$HOME/.dotnet/tools/.store/powershell/7.2.9/powershell/7.2.9/tools/net6.0/any/unix`

**Comment by BethanyZhou** [comment]
Thanks @JamesWTruher. Is there any command to get the dotnet tool install location? I tried on ADO mac agent , it's not the same path with you. It's not easy to search ADO agent :( .

**Comment by TravisEz13** [comment]
The fix for this issue is merged into the v7.2.10 branch and is targeted for that release.
PR #19129 is adding verification to prevent this from regressing in the future.

**Comment by BethanyZhou** [comment]
When 7.2.10 will be available?

**Comment by BethanyZhou** [comment]
After I copied $PSHOME/Microsoft.ApplicationInsights.dll to dotnet tool installation path. Following warning and errors show up:

WARNING: InternalWebProxy: Could not load file or assembly 'Microsoft.CodeAnalysis, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. An attempt was made to load a program with an incorrect format.

Import-Module: Could not load file or assembly 'System.Private.ServiceModel, Version=4.9.0.0,
Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. An attempt was made to load
a program with an incorrect format.

Import-Module: Could not load file or assembly 'System.Configuration.ConfigurationManager,
Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'.

Import-Module: Could not load file or assembly 'System.Management, Version=4.0.0.0,
Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'.

Will stick on pwsh7.2.8 for the time being, please let me know when 7.2.10 is available.
